packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -153,17 +153,22 @@ class SectionHeader extends React.Component {
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     /* eslint-disable jsx-a11y/no-static-element-interactions */
 
-    const buttonAttributes = (onClick) ? { role: 'button', 'aria-expanded': isOpen, 'aria-label': headerText } : undefined;
+    const buttonAttributes = (onClick) ? {
+      type: 'button', tabIndex: '-1', 'aria-expanded': isOpen, 'aria-label': headerText,
+    } : undefined;
+    const spanAttributes = { tabIndex: '-1' };
```

Comment:
Right now it is possible (if onClick=true and accordionIcon=false) to have  an `ArrangeWrapperTag` that is a `span` with `buttonAttributes` and `spanAttributes`. Is this intended or should only these two cases:

1. `ArrangeWrapperTag` is a `button` with `buttonAttributes`
2. `ArrangeWrapperTag` is a `span` with `spanAttributes`

be allowed?

Reply:
We have two section headers one is plain section header (without toggle button) which will render inside span tag and another one is section header with toggle button which will render with button tag. So by using the "onClick" and "accordionIcon" conditions we were rendering button and span elements. We have removed the variable "spanAttributes" and added the corresponding attribute "tabIndex: -1" to else condition of buttonAttributes variable. And also updated the variable name "buttonAtrributes" to "elementAttributes". We have added "ArrangeWrapperTag" which was suggested by saad on the previous comment. https://github.com/cerner/terra-core/pull/3831#discussion_r1256454357